CarsAndBids.Core/CQRS/Cars/DeleteCarByIdCommand.cs

```diff
@@ -17,9 +19,11 @@ IFileService fileService
 {
     public async Task Handle(DeleteCarByIdCommand cmd, CancellationToken cancellationToken)
     {
-        var urls = (await carImageRepository.GetAsync(filter: ci => ci.CarId == cmd.Id))
-            .Select(ci => ci.ImageUrl)
-            .ToList();
+        var filter = (Expression<Func<CarImage, bool>>)(ci => ci.CarId == cmd.Id);
```

Comment:
Create a specification in a separate class and use it here (like I sent you an example )

Reply:
However you can keep this one, but looks as too many line of codes... try to decrease them